presto-orc/src/main/java/com/facebook/presto/orc/reader/StructStreamReader.java

```diff
@@ -293,6 +291,9 @@ public long getRetainedSizeInBytes()
 
     private void setupForScan()
     {
+        if (filter != null && (filter instanceof Filters.IsNull || filter instanceof Filters.AlwaysFalse)) {
```

Comment:
this change doesn't seem to belong to this PR; the logic itself seems to belong to the caller